### PR TITLE
Time and threads clib bindings implemented

### PR DESCRIPTION
**Pull request by markehammons**

### CODE REVIEW

File: clib/src/main/scala/scala/scalanative/libc/stdlib.scala [hunk 1]
```diff
@@ -2,10 +2,13 @@ package scala.scalanative
 package libc
 
 import scalanative.unsafe._
+import scala.scalanative.unsigned.ULong
 
 @extern
 object stdlib {
 
+  type size_t = ULong
```
Comment (LeeTibbert) [inline clib/src/main/scala/scala/scalanative/libc/stdlib.scala:10]: Picky, picky: Please consider `type size_t = CSize"
That has better, to me, semantics. You can also then drop
an import.

In a better world, then the rest of the file would use `size_t` but I understand
the virtue of respecting prior art.

**Comment by LeeTibbert** [comment]
Mark,  Thank you for this work.  A couple of thoughts after reading through.
I mean to be supportive, not picky.

1) My constant question is "What standard or existing practice does this code implement"?
    Without a clear understanding of which standard, I, for one can have no idea of names,
     arguments, completeness, etc.   Obviously it is not POSIX. 

     The existing_code/prior_art is pretty silent about what it is implementing against. Adding
     new code gives a chance to correct that for the introduced code.  Some devo down the 
     line, and more likely more than one, will sing your praises.     

     See below about cppreference URL.

  
   2) I did not  see  `docs/lib/libc.rst` updated to list the new routines.  Did I miss something?
       At the top of that file is the block:
```
Scala Native provides bindings for a core subset of the
`C standard library <https://en.cppreference.com/w/c/header>`_:
```
I find those descriptions pretty useless, but that is the art.  What I am proposing
in #1 above is that a statement in the .c (.scala?) file that the code if following
that guidon.    That puts the implementation and the intended standard in
front of the eyes of someone reading the file. It reduces the required lore
and the need make a trip to ReadTheDocs.


3) I am in the minority,  but I have always found it useful to have some form of Test,
    
    This ensures that arguments are correct, things link, etc.   I concur that the
    underlying functionality need not be extensively tested, just activated/touched.
    Of course, this is within the limits of what JUnit will allow.  I personally find
    `scripted` burdensome beyond value returned.

**Comment by LeeTibbert** [comment]
I think the idea/hope is that someday `posixlib` will call into an underlying libc implementation, where it
can, rather than having all sorts of duplicated code, with different bugs, lying around.
Of course POSIX CX and XSI probably can not do this.  

Such a world would give a cleaner factorization.

**Comment by markehammons** [comment]
Hi @LeeTibbert I'll make the changes requested soon. The documentation changes are a bit tougher than the rest, as are the testing changes.

As to what standard is this following, it's [C11](https://devdocs.io/c/thread). This API is sadly rather unimplemented (even clang 6 doesn't support it), so despite being part of the c standard lib, its availability will depend on the platform and clang version the user is using. I've added guards in the c bindings to make sure they are not compiled in the case this piece of the C11 standard is missing or unsupported.

**Comment by catap** [comment]
@markehammons keep in mind that just implementing threads isn't enough. You need to introduce it in GC.

JFI: I'm working (really hard) to introduce a support of real multithreading that was started long ago. Here a two branches that I'm using to work:
 - https://github.com/catap/scala-native/tree/multithreading
 - https://github.com/catap/scala-native/tree/concurrent-immix

Anyway, it contains major issues like it is leaking :) that is prevent me to open a PR.

**Comment by ekrich** [comment]
My recommendation is that you have a PR for each header file that you want to add or make changes to as shown here: https://en.cppreference.com/w/c/header

This will make the changes much smaller and the review much easier.

**Comment by WojciechMazur** [comment]
Closing as a stale PR

**Pull request closed without merging**